collab2/foraging/toolkit/decay.py

```diff
@@ -0,0 +1,27 @@
+import copy
+from typing import Callable, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from collab2.foraging.toolkit import filter_by_distance
+from collab2.foraging.toolkit.utils import dataObject  # noqa: F401
+
+
+def _exponential_decay(distance: Union[float, np.ndarray], 
+                      decay_factor: float = 0.5):
+    return np.exp(-decay_factor * distance)
+
+
+def _decaying_contribution(
+    x_source: int,
+    y_source: int,
+    grid: pd.DataFrame,
+    decay_function: Callable = _exponential_decay,
+    **decay_function_kwargs,
+) -> np.ndarray:
+  
+
+    distance_to_source = np.sqrt((grid["x"] - x_source) ** 2 + (grid["y"] - y_source) ** 2)
+    proximity_score = decay_function(distance_to_source, **decay_function_kwargs)
```

Comment:
if you are thinking of using _decaying_contribution as something generic to be used for proximity/food/ any other exponential, then I would rename `proximity_score` to something more general like `predictor_contribution`

Reply:
done more re-factorinig, to make this even more general, had impact on proximity